intermediate-js.md

```diff
@@ -1,39 +1,73 @@
 # Intermediate Javascript Assessments
 
 ### Without using Google answer the following:
-
 1. What is the difference between .map() and .filter()?
 
+  .filter() takes an array and filters certain elements from that array into another array. I think about .filter() as a forEach which uses .pop() to remove elements that meet a criteria.
+  .map() takes an array and mutates certain values and puts them into another array. Its a forEach that changes values.
+
 2. Why would you use object destructuring?
 
+  Object destructuring allows for the coder to call on object keys/values without having to call on the object itself. It makes code earsier to read/write.
+
 3. What is the difference between var, let, and const?
 
+  The main difference between var, let, and const is scope. Const is used to define variables that are not meant to be changed. Let is used usually in a local scope to define variables that are dynamic. Var is used for cases inbetween.
+
 4. Why is testing important?
 
+  Testing is important for a number of reasons. Users and investors gain trust in companies who test their code. Testing helps developers check small blocks of code to make sure they work. Testing may also help the developer understand what is needed to code in the future to add new/necessary functionality.
+
 5. What is a higher order function?
 
+  A function that takes in or outputs another function.
+
 6. What is the difference between a class and an object?
 
+  An object in javascrpt has key and value pairs as well as behavior. Classes in javascript used to construct objects.
+
 7. What did you learn during the group project this week? Please include any additional feedback you may have.
 
+  Battleship was a huge step for me this week. I am starting to understand the full potential for React. Working in a group for this project was difficult because there were so many ways of implementing our ideas. At times it was frustrating but overall a great experience.
+
 
 ### HTML/CSS Review questions: First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. How do you link a CSS file to your HTML page?
 
+  <script src="file_path_goes_here.css" />
+
 2. What is the difference between a div and a span?
 
+  A div is used to nest html elements. Span is used to group elements in order to style or perform js on them.
```

Comment:
Looks like you've got an idea of why you might use divs, which is a good start. Do bit of research into divs and spans and see what you come up with.